test/com/serotonin/mango/vo/report/CalculateConsolidatedChartHeightTest.java

```diff
@@ -0,0 +1,263 @@
+package com.serotonin.mango.vo.report;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(Parameterized.class)
+public class CalculateConsolidatedChartHeightTest {
+
+    private final List<ReportChartCreator.PointStatistics> pointStatistics;
+    private final int expectedHeight;
+    private final int imageHeightForDataPointNameInLegend;
+    private final int imageHeight;
+    private final int charAmountPerLine;
+    public CalculateConsolidatedChartHeightTest(List<ReportChartCreator.PointStatistics> pointStatistics, int expectedHeight, int imageHeightForDataPointNameInLegend, int imageHeight, int charAmountPerLine) {
+        this.pointStatistics = pointStatistics;
+        this.expectedHeight = expectedHeight;
+        this.imageHeightForDataPointNameInLegend = imageHeightForDataPointNameInLegend;
+        this.imageHeight = imageHeight;
+        this.charAmountPerLine = charAmountPerLine;
+    }
+
+    @Parameterized.Parameters(name = "{index}: calculateConsolidatedChartHeight(points: {0}, expected height: {1},pixels per line: {2} base height of chart: {3}, characters per line: {4})")
+    public static Collection<Object[]> data() {
+        String _0CharactersPointName = "";
```

Comment:
A local variable should not be named after the _ character, in some sources this character is used to denote private fields and methods.
The name may look like this:
 chars0PointName